File: VB/Grid_DateOccurring/Form1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.XtraEditors

Namespace Grid_DateOccurring
	Partial Public Class Form1
		Inherits DevExpress.XtraEditors.XtraForm

		Public Sub New()
			InitializeComponent()

			dashboardViewer1.DataSourceOptions.ObjectDataSourceLoadingBehavior = DevExpress.DataAccess.DocumentLoadingBehavior.LoadAsIs

			Dim dashboard As New Dashboard()
			dashboard.LoadFromXml("..\..\Data\Dashboard.xml")
			dashboardViewer1.Dashboard = dashboard
			Dim grid As GridDashboardItem = CType(dashboard.Items("gridDashboardItem1"), GridDashboardItem)
			Dim [date] As GridDimensionColumn = CType(grid.Columns(0), GridDimensionColumn)

			Dim dateOccurringRule As New GridItemFormatRule([date])
			Dim dateOccurringCondition As New FormatConditionDateOccurring()
			dateOccurringCondition.DateType = FilterDateType.MonthAgo1 Or FilterDateType.MonthAgo2
			dateOccurringCondition.StyleSettings = New AppearanceSettings(FormatConditionAppearanceType.PaleOrange)
			dateOccurringRule.Condition = dateOccuringCondition
			dateOccurringRule.ApplyToRow = True

			grid.FormatRules.Add(dateOccurringRule)
		End Sub
	End Class
End Namespace

File: CS/Grid_DateOccurring/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.XtraEditors;

namespace Grid_DateOccurring {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        public Form1() {
            InitializeComponent();

            dashboardViewer1.DataSourceOptions.ObjectDataSourceLoadingBehavior = DevExpress.DataAccess.DocumentLoadingBehavior.LoadAsIs;

            Dashboard dashboard = new Dashboard();
            dashboard.LoadFromXml(@"..\..\Data\Dashboard.xml");
            dashboardViewer1.Dashboard = dashboard;
            GridDashboardItem grid = (GridDashboardItem)dashboard.Items["gridDashboardItem1"];
            GridDimensionColumn date = (GridDimensionColumn)grid.Columns[0];

            GridItemFormatRule dateOccuringRule = new GridItemFormatRule(date);
            FormatConditionDateOccurring dateOccuringCondition = new FormatConditionDateOccurring();
            dateOccuringCondition.DateType = FilterDateType.MonthAgo1 | FilterDateType.MonthAgo2;
            dateOccuringCondition.StyleSettings = 
                new AppearanceSettings(FormatConditionAppearanceType.PaleOrange);
            dateOccuringRule.Condition = dateOccuringCondition;
            dateOccuringRule.ApplyToRow = true;

            grid.FormatRules.Add(dateOccuringRule);            
        }
    }
}
